File: benchmarks/comparison/sieve.py
def is_prime(n):
    if n < 2:
        return False
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    return True

count = 0
for n in range(2, 5000000):
    if is_prime(n):
        count += 1
print(count)

File: benchmarks/comparison/sieve.rs
fn is_prime(n: i64) -> bool {
    if n < 2 { return false; }
    let mut i: i64 = 2;
    while i * i <= n { if n % i == 0 { return false; } i += 1; }
    true
}

fn main() {
    let mut count: i64 = 0;
    for n in 2i64..5_000_000 {
        if is_prime(n) { count += 1; }
    }
    println!("{}", count);
}
